src/storage/txn/latch.rs

```diff
@@ -86,33 +144,32 @@ impl Latches {
         H: Hash,
     {
         // prevent from deadlock, so we sort and deduplicate the index
-        let mut slots: Vec<usize> = keys.iter().map(|x| self.calc_slot(x)).collect();
-        slots.sort();
-        slots.dedup();
-        Lock::new(slots)
+        let mut hashes: Vec<u64> = keys.iter().map(|x| self.calc_slot(x)).collect();
+        hashes.sort();
```

Comment:
Sorting the hashes cannot prevent deadlock. For example, we have 1024 slots. Command A has key hashes `[0, 1025]` while command B has key hashes `[1, 1024]`. Then there can be a deadlock.
So here I believe you need to sort by slot id.

Reply:
Good. This is a bug.

Reply:
No such problem, since each slot actually can handle mutiple different hashes.

Reply:
Each lock only can hold one mutex at a time.
https://github.com/tikv/tikv/pull/6094/files#diff-56bcb7f504d93784d808674ed821bdbaR161-R162

Reply:
Ah, sorry. Never mind. It's ok to sort by key hash.